zebra-chain/src/sapling/commitment.rs

```diff
@@ -120,11 +123,9 @@ impl NoteCommitment {
         // The `TryFrom<Diversifier>` impls for the `jubjub::*Point`s handles
         // calling `DiversifyHash` implicitly.
 
-        let g_d_bytes: [u8; 32] = if let Ok(g_d) = jubjub::AffinePoint::try_from(diversifier) {
-            g_d.to_bytes()
-        } else {
-            return None;
-        };
+        let g_d_bytes = jubjub::AffinePoint::try_from(diversifier)
+            .map_err(|_| CryptoError::Affine(AffinePointError::FromDiversifier))?
```

Comment:
`AffinePointError` as a top-level `Error` is a bit ambiguous as we have several curve point types that can be represented in that manner; I would suggest having a `CryptoError::InvalidPoint` type that an `InvalidPointError` variant can be converted from